hack/tools/deployment/deployment-update.go

```diff
@@ -0,0 +1,295 @@
+		log.Printf("MR Description as follows\n\n%s", string(cmdOutput))
+	}
+}
+
+// updateDeploymentByEnvironment updates the sectors in a particular ROSA environment
+func updateDeploymentByEnvironment(env string, lines []string, oldTag, newTag, newCommit string) error {
+	for _, sector := range ROSAEnvironments[env] {
+		err := updateTagAndCommitSHA(&lines, sector, oldTag, newTag, newCommit)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// isValidEnvironment verifies the environment passed through the flag is valid
+func isValidEnvironment(env string) bool {
+	_, valid := ROSAEnvironments[env]
+	return valid
+}
+
+// updateTagAndCommitSHA replaces the git tag in one line and the commit sha in the following line in the deployment file
+//
+// Example lines containing the git tag and the commit sha
+// # openshift/hypershift.git git-branch: main, git-tag: v0.1.32, git-commit: image tag
+// hypershift-operator: quay.io/acm-d/rhtap-hypershift-operator:9e84188
+func updateTagAndCommitSHA(lines *[]string, sector, oldTag, newTag, newCommit string) error {
+	found := false
+	modified := false
+	fileContent := *lines
+
+	for i, line := range fileContent {
+		// Once the sector is found, keep reading each line until we hit the line with the git tag
+		if found {
+			if strings.Contains(line, oldTag) {
+				// Replace the tag in the line
+				fileContent[i] = strings.Replace(line, oldTag, newTag, 1)
+
+				// The next line should be the commit; update the commit
+				commitStrings := strings.Split(fileContent[i+1], ":")
+				if len(commitStrings) != 3 {
+					return fmt.Errorf("invalid commit format")
+				}
+
+				commitStrings[2] = newCommit
+				fileContent[i+1] = strings.Join(commitStrings, ":")
+				modified = true
+				log.Printf("Updated sector %s", sector)
+
+				break
+			}
+
+			// We're starting into another sector, so we need to break
+			if strings.Contains(line, "  - name: ") {
+				break
+			}
+		}
+
+		// We found the sector we want to update
+		if strings.Contains(line, sector) {
+			found = true
+		}
+	}
+
+	if !found {
+		log.Printf("could not find the sector in the file: %s", sector)
+	}
+
+	if found && !modified {
+		log.Printf("did not find a line containing the old tag, %s, for sector '%s'", oldTag, sector)
+	}
+
+	return nil
+}
+
+func gitCheckout(repoPath, branch string) error {
+	cmd := exec.Command("git", "checkout", branch)
+	cmd.Dir = repoPath
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("error checking out branch: %s\n%s", err, output)
+	}
+	fmt.Printf("Output: %s\n", output)
+	return nil
+}
+
+func gitCreateBranch(repoPath, branch string) error {
+	cmd := exec.Command("git", "checkout", "-b", branch)
+	cmd.Dir = repoPath
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("error creating branch: %s\n%s", err, output)
+	}
+	fmt.Printf("Output: %s\n", output)
+	return nil
+}
+
+func gitAdd(repoPath string) error {
+	cmd := exec.Command("git", "add", ".")
```

Comment:
Prefer to add specific files in automated tools to reduce the chances of accidental inclusion